commit 77ca7286d10b798e4907af941f29672bf484db77
Author: Mike Miller <mike.miller@hp.com>
Date:   Thu Nov 6 12:53:14 2008 -0800

    cciss: new hardware support
    
    Add support for 2 new SAS/SATA controllers.
    
    Signed-off-by: Mike Miller <mike.miller@hp.com>
    Cc: Jens Axboe <jens.axboe@oracle.com>
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

Adjusted to apply to Debian's 2.6.18 by dann frazier <dannf@debian.org>

diff -urpN a/Documentation/cciss.txt b/Documentation/cciss.txt
--- a/Documentation/cciss.txt	2008-12-23 12:21:29.000000000 -0700
+++ b/Documentation/cciss.txt	2008-12-23 12:22:54.000000000 -0700
@@ -26,6 +26,8 @@ This driver is known to work with the fo
 	* SA P410i
 	* SA P411
 	* SA P812
+	* SA P712m
+	* SA P711m
 
 If nodes are not already created in the /dev/cciss directory, run as root:
 
diff -urpN a/drivers/block/cciss.c b/drivers/block/cciss.c
--- a/drivers/block/cciss.c	2008-12-23 12:21:29.000000000 -0700
+++ b/drivers/block/cciss.c	2008-12-23 12:22:54.000000000 -0700
@@ -90,6 +90,8 @@ static const struct pci_device_id cciss_
 	{PCI_VENDOR_ID_HP,     PCI_DEVICE_ID_HP_CISSE,     0x103C, 0x3245},
 	{PCI_VENDOR_ID_HP,     PCI_DEVICE_ID_HP_CISSE,     0x103C, 0x3247},
 	{PCI_VENDOR_ID_HP,     PCI_DEVICE_ID_HP_CISSE,     0x103C, 0x3249},
+	{PCI_VENDOR_ID_HP,     PCI_DEVICE_ID_HP_CISSE,     0x103C, 0x324A},
+	{PCI_VENDOR_ID_HP,     PCI_DEVICE_ID_HP_CISSE,     0x103C, 0x324B},
 	{PCI_VENDOR_ID_HP,     PCI_ANY_ID,	PCI_ANY_ID, PCI_ANY_ID,
 		PCI_CLASS_STORAGE_RAID << 8, 0xffff << 8, 0},
 	{0,}
@@ -127,6 +129,8 @@ static struct board_type products[] = {
 	{0x3245103C, "Smart Array P410i", &SA5_access},
 	{0x3247103C, "Smart Array P411", &SA5_access},
 	{0x3249103C, "Smart Array P812", &SA5_access},
+	{0x324A103C, "Smart Array P712m", &SA5_access},
+	{0x324B103C, "Smart Array P711m", &SA5_access},
 	{0xFFFF103C, "Unknown Smart Array", &SA5_access},
 };
 
